cooperator/models/company.py

```diff
@@ -138,3 +178,78 @@ def onchange_financial_risk_approval_required(self):
     def onchange_generic_rules_approval_required(self):
         if self.generic_rules_approval_required:
             self.display_generic_rules_approval = True
+
+    @api.model
+    def create(self, vals):
+        result = super().create(vals)
+        # The ignore_list is populated such that, if the user defines a template
+        # they want to use during company creation, that choice doesn't get
+        # overwritten. The boolean check exists because, when creating a company
+        # from the UI, all empty fields are defined with no value.
+        result._setup_default_cooperator_mail_templates(
+            ignore_list=[key for key, val in vals.items() if val]
+        )
+        return result
+
+    @api.model
+    def _get_cooperator_mail_template_fields(self):
+        return {
+            "cooperator_confirmation_mail_template": "cooperator.email_template_confirmation",
+            "cooperator_confirmation_company_mail_template": (
+                "cooperator.email_template_confirmation_company"
+            ),
+            "cooperator_capital_release_mail_template": (
+                "cooperator.email_template_release_capital"
+            ),
+            "cooperator_waiting_list_mail_template": "cooperator.email_template_waiting_list",
+            "cooperator_certificate_mail_template": "cooperator.email_template_certificat",
+            "cooperator_certificate_increase_mail_template": (
+                "cooperator.email_template_certificat_increase"
+            ),
+            "cooperator_share_transfer_mail_template": (
+                "cooperator.email_template_share_transfer"
+            ),
+            "cooperator_share_update_mail_template": "cooperator.email_template_share_update",
+        }
+
+    def _get_default_cooperator_mail_template(self, xmlid, copy=True):
+        """Get the mail template from its xmlid and either return it or return
+        a copy of it.
+        """
+        self.ensure_one()
+        try:
+            template = self.env.ref(xmlid)
+        except ValueError:
+            return False
+        if copy:
+            result = template.copy()
+            # Set name to 'Company - Template' instead of 'Template (copy)'.
+            result.name = "{} - {}".format(self.name, template.name)
+            return result
+        else:
+            return template
+
+    def _assign_default_cooperator_mail_template(
+        self, field, xmlid, copy=True, overwrite=False
+    ):
+        for company in self:
+            if overwrite or not getattr(company, field):
+                company.write(
+                    {
+                        field: company._get_default_cooperator_mail_template(
+                            xmlid, copy=copy
+                        ).id
+                    }
+                )
+
+    def _setup_default_cooperator_mail_templates(
```

Comment:
Do we need all this post-migration methods inside the company module? I see we only really use them in the post-migration, maybe we can move them inside, so it looks clearer why we use this.